### PR TITLE
add docstrings to metrics.py and NeuralNet.py

### PR DESCRIPTION
**Pull request by manonreau**

### CODE REVIEW

File: graphprot/Metrics.py [hunk 1]
```diff
@@ -55,22 +111,19 @@ def __init__(self, prediction, y, target, threshold=4, binary=True):
...
             false_positive, false_negative, true_positive, true_negative = get_comparison(
-                prediction_bool, y_bool, self.binary, classes=classes)
+                prediction_binary, y_binary, self.binary, classes=classes)
             
         else:
-            if self.target == 'class':
+            if self.target == 'capri_class':
                 classes = [1, 2, 3, 4, 5]
-            else:
+            if self.target == 'binclass':
```
Comment (NicoRenaud) [inline graphprot/Metrics.py:125]: It's always better to have a default case (that potentially raise an error)
```python
if target == 'capri':
     classes = 
elif target == 'binclass':
      classes = 
else:
      raise ValueError('target must be capri on binclass')
``` 

so that if someday we by accident set target to `bin_class` instead of `binclass` we know what's happening

Comment (NicoRenaud) [reply graphprot/Metrics.py:125]: It's also always better to be as consistent as possible in the names. So if we have `capri_class`  I would suggest we use `bin_class`.

File: graphprot/Metrics.py [hunk 2]
```diff
@@ -160,31 +213,41 @@ def __init__(self, prediction, y, target, threshold=4, binary=True):
...
+        """
         idx, ground_truth_bool = self.format_score()
         return np.cumsum(ground_truth_bool[idx])
 
     def auc(self):
-        
+        """the Receiver Operating Characteristic (ROC) area under the curve (AUC)
+
+        Returns:
+            [type]: [description]
```
Comment (NicoRenaud) [inline graphprot/Metrics.py:250]: no description of the return :(

File: graphprot/NeuralNet.py [hunk 3]
```diff
@@ -123,7 +123,11 @@ def __init__(self, database, Net,
         self.valid_loss = []
 
     def plot_loss(self, name=''):
-        """Plot the loss of the model."""
+        """Plot the loss of the model
+
+        Args:
+            name (str, optional): name of the output file. Defaults to ''.
```
Comment (NicoRenaud) [inline graphprot/NeuralNet.py:129]: that's the epoch number right ?

File: graphprot/NeuralNet.py [hunk 4]
```diff
@@ -146,8 +150,12 @@ def plot_loss(self, name=''):
             plt.close()
 
     def plot_acc(self, name=''):
-        """Plot the accuracy of the model."""
+        """Plot the accuracy of the model.
 
+        Args:
+            name (str, optional): name of the output file. Defaults to ''.
```
Comment (NicoRenaud) [inline graphprot/NeuralNet.py:156]: same as above

**Review by NicoRenaud — APPROVED** [review]
Nice doc strings !! A few very minor remarks

File: graphprot/NeuralNet.py [hunk 5]
```diff
@@ -487,28 +507,35 @@ def get_metrics(self, data='eval', threshold=4, binary=True):
             if len(self.valid_out) == 0:
```
Comment (NicoRenaud) [inline graphprot/NeuralNet.py:507]: is `valid_out` properly initialized in case we don't have a validation set ? I can't find that ... 
same for `test_out` and `train_out`. If they are properly initialized just ignore that comment

**Review by NicoRenaud — APPROVED** [review]
Good work :) 
Some minor remarks and one concern about the initalization of valid_out train_out and test_out.
Unrelated but we should delete the branches that have been merged already :)

File: graphprot/NeuralNet.py (hunk 3 above)
Comment (manonreau) [reply graphprot/NeuralNet.py:129]: No the name is not the epoch since the plot is the loss as a function of the epoch.
The plot can be computed after the model is trained, and the name can be any input from the user.

ex:
```
model.train(nepoch=epoch, validate=True)
model.plot_loss('_fold{}_b{}_e{}_t{}_y{}_shuffle_onehot_EGAT'.format(fold, batch_size, epoch, task, target))

```
